prototype1/src/App.tsx

```diff
@@ -119,19 +118,38 @@ function drawBubbleChart(
 
   const circle = group.append("circle")
 
-  const colormap = new ColdMapTranslator(data.minNoCommits, data.maxNoCommits);
+  const VIEW = {
+    TRUCKFACTOR: 1,
+    FILETYPE: 2,
+    HEATMAP: 3,
+    COLDMAP: 4,
+  }
+
+  function getColorOfBlobDepedentOnViewAdapter(blob: HydratedGitBlobObject, view: number): string {
+    switch (view) {
+      case VIEW.TRUCKFACTOR:
+        return getDominanceColor(blob)
+      case VIEW.FILETYPE:
+        // TODO implement filetype color function
+        throw new Error(`Color function for view: VIEW.FILETYPE is not yet implemented`);
+      case VIEW.HEATMAP:
+        return new HeatMapTranslator(data.minNoCommits, data.maxNoCommits).getColor(blob)
```

Comment:
Could these class instances be saved, instead of reconstructed each time `getColorOfBlobDepedentOnViewAdapter` is called? The data variable is accessed from outside anyways.